packages/dataviz/src/LineChart/LineChart.component.js

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {
+	Legend,
+	LineChart,
+	Line,
+	CartesianGrid,
+	XAxis,
+	YAxis,
+	Tooltip,
+	ResponsiveContainer,
+} from 'recharts';
+import {
+	isSelectedOrHighlight,
+	getActiveDotR,
+	getDotR,
+	getStrokeOpacity,
+	getStrokeWidth,
+	union,
+} from './linechart-utils';
+
+import './LineChart.scss';
+
+export function handleMouseDown(selectedLegends, setSelectedLegends) {
+	return o => {
+		const { dataKey } = o;
+		if (!selectedLegends.find(legend => legend === dataKey)) {
+			setSelectedLegends(selectedLegends.concat([dataKey]));
+		} else {
+			setSelectedLegends(selectedLegends.filter(legend => legend !== dataKey));
+		}
+	};
+}
+
+export function handleMouseEnter(setHighlightLegendKey) {
+	return o => {
+		const { dataKey } = o;
+		setHighlightLegendKey(dataKey);
+	};
+}
+
+export function handleMouseLeave(setHighlightLegendKey) {
+	return () => {
+		setHighlightLegendKey('');
+	};
+}
+
+export function renderLine(data, key, highlightLegendKey, selectedLegends, colors) {
+	let dataKeys = new Set();
+	data.forEach(dots => {
+		dataKeys = union(dataKeys, new Set(Object.keys(dots)));
+	});
+	dataKeys.delete(key);
+	return [...dataKeys].map((dataKey, index) => {
```

Comment:
way too much operations here, I m sure we can do more efficient.
We first create a set, then three sets are created for every data. to finaly create a new Array from the set to call map on it.

I believe a simple reduce of data in a Map and them just map over the keys.